cache/src/lib.rs

```diff
@@ -280,24 +296,22 @@ impl Cache {
 }
 
 pub async fn accessible_to_cache_item(accessible: &AccessibleProxy<'_>) -> OdiliaResult<CacheItem> {
-	let (app, parent, index, children, ifaces, role, states, text) = tokio::try_join!(
+	let (app, parent, index, children, ifaces, role, states) = tokio::try_join!(
 		accessible.get_application(),
 		accessible.parent(),
 		accessible.get_index_in_parent(),
 		accessible.child_count(),
 		accessible.get_interfaces(),
 		accessible.get_role(),
 		accessible.get_state(),
-		async {
-			// if it implements the Text interface
-			match accessible.to_text().await {
-				// get *all* the text
-				Ok(text_iface) => text_iface.get_text_ext().await,
-				// otherwise, use the name instaed
-				Err(_) => accessible.name().await,
-			}
-		},
 	)?;
+	// if it implements the Text interface
+	let text = match accessible.to_text().await {
+		// get *all* the text
+		Ok(text_iface) => text_iface.get_all_text().await,
+		// otherwise, use the name instaed
+		Err(_) => Ok(accessible.name().await?),
+	}?;
```

Comment:
Doesn't need to hold up this PR as I can revise it in the main one, but is there a good reason to pull this `text` out of the `try_join`? It doesn't look like it depends on any of the things above, so I would think it should remain concurrent.

Reply:
Incompatible Result types.

Reply:
Ah, I see. we could map all of them `into` the `OdiliaResult` in each branch to keep them concurrent, it just gets a tad messy with now having to `async { _.await }` in each one. I'll do this at some point :+1: